core/src/main/java/jenkins/install/SetupWizard.java

```diff
@@ -122,6 +123,10 @@ public SetupWizard() {
                     authStrategy.setAllowAnonymousRead(false);
                     jenkins.setAuthorizationStrategy(authStrategy);
     
+                    // Disable the legacy system of API Token
+                    ApiTokenPropertyConfiguration.get().setCreationOfLegacyTokenDisabled(true);
+                    ApiTokenPropertyConfiguration.get().setTokenGenerationOnCreationDisabled(true);
```

Comment:
Do you have another possibility to avoid a fresh install to show warnings from monitor even after a run of the wizard because they are enabled by default ? (the change from negative to positive approach does not impact this behavior).

IIUC one of the wizard's goal is to provide "good" defaults.

Reply:
Would be better to disable legacy by default and figure out whether it's an upgrade from earlier, and only enable it in that case.

Not sure what the best way is, @dwnusbaum and @jglick might have ideas as they've recently worked on initialization IIUC.

(And in general it's probably useful for some of the existing options to be moved out of the wizard. Perhaps we can keep it here but plan to have a followup change that changes the default options and instead detects an update?)

Reply:
Regarding enabling on upgrade, I think you could put the logic in `UpgradeWizard.java#initializeState`, which corresponds to `InstallState.UPGRADE`, but it would need to be tested.

Reply:
@dwnusbaum thank you for the tip, I will test that :)

Reply:
I tested the existence of the config file of `ApiTokenPropertyConfiguration` during the upgrade process to know if I must apply default or not.